src/text-editor-registry.js

```diff
@@ -214,14 +218,40 @@ class TextEditorRegistry {
     atom.grammars.autoAssignLanguageMode(editor.getBuffer())
   }
 
-  async subscribeToSettingsForEditorScope (editor) {
+  updateEditorSettingsForLanguageMode(editor, oldLanguageMode) {
+    const newLanguageMode = editor.buffer.getLanguageMode()
+
+    if (oldLanguageMode) {
```

Comment:
I'm surprised that `oldLanguageMode` would ever be null. When does that happen?

Reply:
In [this call](https://github.com/atom/atom/pull/16542/files#diff-c07ba0f74b082a7f8c36ffec1a1ef1b8R152) that establishes the settings of the editor in `maintainConfig`.  Maybe there's a cleaner way to accomplish the same thing?